@xen-orchestra/lite/src/components/pool/network/PoolHostInternalNetworkTable.vue

```diff
@@ -0,0 +1,202 @@
+      />
+    </div>
+  </div>
+  <UiCardSpinner v-if="!isReady" />
+</template>
+
+<script setup lang="ts">
+import UiCardSpinner from '@/components/ui/UiCardSpinner.vue'
+import useMultiSelect from '@/composables/multi-select.composable'
+import type { XenApiNetwork } from '@/libs/xen-api/xen-api.types'
+import VtsIcon from '@core/components/icon/VtsIcon.vue'
+import ColumnTitle from '@core/components/table/ColumnTitle.vue'
+import VtsTable from '@core/components/table/VtsTable.vue'
+import UiButton from '@core/components/ui/button/UiButton.vue'
+import UiButtonIcon from '@core/components/ui/button-icon/UiButtonIcon.vue'
+import UiCheckbox from '@core/components/ui/checkbox/UiCheckbox.vue'
+import UiQuerySearchBar from '@core/components/ui/query-search-bar/UiQuerySearchBar.vue'
+import UiTableActions from '@core/components/ui/table-actions/UiTableActions.vue'
+import UiTitle from '@core/components/ui/title/UiTitle.vue'
+import UiTopBottomTable from '@core/components/ui/top-bottom-table/UiTopBottomTable.vue'
+import { useTable } from '@core/composables/table.composable'
+import { vTooltip } from '@core/directives/tooltip.directive'
+import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
+import {
+  faAlignLeft,
+  faCaretDown,
+  faEdit,
+  faEllipsis,
+  faHashtag,
+  faPlus,
+  faTrash,
+} from '@fortawesome/free-solid-svg-icons'
+import { computed, ref, watchEffect } from 'vue'
+
+const props = defineProps<{
+  hostInternalNetwork: XenApiNetwork[]
+  isReady: boolean
+}>()
+
+const reactiveHostInternalNetworks = ref<XenApiNetwork[]>(props.hostInternalNetwork || [])
+
+const searchQuery = ref('')
+
+const filteredNetworks = computed(() => {
+  return searchQuery.value
+    ? reactiveHostInternalNetworks.value.filter(network =>
+        Object.values(network).some(value => String(value).toLowerCase().includes(searchQuery.value.toLowerCase()))
+      )
+    : reactiveHostInternalNetworks.value
+})
+
+const usableRefs = computed(() => reactiveHostInternalNetworks.value.map(item => item.uuid))
+
+const { selected, areAllSelected } = useMultiSelect(usableRefs)
+
+const toggleSelect = () => {
+  selected.value = selected.value.length === 0 ? usableRefs.value : []
+}
+
+const { visibleColumns, rows } = useTable('networks', filteredNetworks, {
+  rowId: record => record.uuid,
+  columns: define => [
+    define('checkbox', () => '', { label: '', isHideable: false }),
+    define('name_label', { label: 'Name', isHideable: true }),
+    define('name_description', { label: 'Description', isHideable: true }),
+    define('MTU', { label: 'MTU', isHideable: true }),
+    define('default_locking_mode', { label: 'Default Locking mode', isHideable: true }),
+    define('more', () => '', { label: '', isHideable: false }),
+  ],
+})
+
+type networkHeader = 'name_label' | 'name_description' | 'MTU' | 'default_locking_mode' | 'more'
+
+const headerIcon: Record<networkHeader, { icon: IconDefinition }> = {
```

Comment:
Types should use `PascalCase`

```suggestion
type NetworkHeader = 'name_label' | 'name_description' | 'MTU' | 'default_locking_mode' | 'more'

const headerIcon: Record<NetworkHeader, { icon: IconDefinition }> = {
```